### PR TITLE
Better documentation creation ref #648

### PR DESCRIPTION
**Pull request by Roba1993**
Hey, 

I implemented the enhancements for the documentation generation #648 by myself.
Looking forward to get these improvements merged 😄 

Right now I called the link back to the root overview just 'Overview', maybe you have a better name.

Please let me know if something if not as you like.
Thanks,
Robert

### CODE REVIEW

**Comment by udoprog** [comment]
Looks very nice!

File: crates/rune/src/doc/static/enum.html.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{#> layout}}
-<h3 class="title">Enum {{literal module}}::<span class="enum">{{name}}</span></h3>
+<div class="title-wrapper"><h3 class="title">Enum {{literal module}}::<span class="enum">{{name}}</span></h3><a class="overview" href="{{literal root_index}}">Overview</a></div>
```
Comment (udoprog) [inline crates/rune/src/doc/static/enum.html.hbs:2]: Could you post a screenshot of how the overview links is expected to render?

**Comment by Roba1993** [comment]
Closes #648 when merged

**Comment by udoprog** [comment]
Thanks for putting this together!

**Pull request merged**